Analysis/interface/TauSelection.h

```diff
@@ -0,0 +1,43 @@
+#include "TauMLTools/Analysis/interface/TauTuple.h"
+#include "TauMLTools/Core/interface/exception.h"
+
+#include "TLorentzVector.h"
+
+namespace analysis {
+    bool PassGenLeptonCut(const tau_tuple::Tau& tau)
+    {   
+        if (tau.genLepton_index >= 0){
+            TLorentzVector genv;
+            TLorentzVector tauv;
+
+            tauv.SetPtEtaPhiM(tau.tau_pt, tau.tau_eta, tau.tau_phi, tau.tau_mass);
+            genv.SetPtEtaPhiM(  tau.genLepton_vis_pt   ,
+                                tau.genLepton_vis_eta  ,
+                                tau.genLepton_vis_phi  ,
+                                tau.genLepton_vis_mass );
+            
+            if (tauv.DeltaR(genv) > 0.2){
+                return false;
+            }
+
+            if (tau.genLepton_kind > 0 && tau.genLepton_kind < 5){
```

Comment:
To be more robust in case of possible changes of the definitions, better to cast `genLepton_kind` to `reco_tau::gen_truth::GenLepton::Kind` and explicitly compare enum values.